core/src/main/scala/kafka/controller/ControllerRequestMerger.scala

```diff
@@ -135,13 +151,18 @@ class ControllerRequestMerger(config: KafkaConfig) extends Logging {
     }}
     leaderAndIsrLiveLeaders = request.liveLeaders()
     leaderAndIsrCallback = callback
+    // we should stop the merging when a full LeaderAndISR request is encountered
+    !isFullLeaderAndIsr
```

Comment:
Not that you need to change it, but is this condition slightly stricter than necessary? That is, could merging instead continue up until the next LAIR is encountered, and stop just before that one?

Reply:
You are right, it's stricter than necessary, and it's valid to continue merging until the next LAIR.
Since the full LAIR is only sent only once for each broker and doesn't cause perf concerns,
I feel having this stricter condition is easier to reason about.

Reply:
Yup, agreed. Just wanted to make sure my understanding was correct.